mypyc/irbuild/ll_builder.py

```diff
@@ -656,10 +656,14 @@ def compare_tuples(self,
             lhs_item = lhs_items[i]
             rhs_item = rhs_items[i]
             compare = self.binary_op(lhs_item, rhs_item, op, line)
+            # Cast to bool if necessary since most types uses comparison returning a object type
+            # See generic_ops.py for more information
+            if not is_bool_rprimitive(compare.type):
+                comapre = self.coerce(compare, bool_rprimitive, line)
```

Comment:
typo ("comapre")

Reply:
Thanks! But I've already addressed that in e34c025. The problem now is the `Tuple[T, ...]` nested within a regular Tuple